docs/handbook/image-file-formats.rst

```diff
@@ -106,8 +106,27 @@ writes run-length encoded files in GIF87a by default, unless GIF89a features
 are used or GIF89a is already in use.
 
 GIF files are initially read as grayscale (``L``) or palette mode (``P``)
-images, but seeking to later frames in an image will change the mode to either
-``RGB`` or ``RGBA``, depending on whether the first frame had transparency.
+images. Seeking to later frames in a ``P`` image will change the image to
+``RGB`` (or ``RGBA`` if the first frame had transparency).
+
+``P`` mode images are changed to ``RGB`` because each frame of a GIF may
+introduce up to 256 colors. Because ``P`` can only have up to 256 colors, the
+image is converted to handle all of the colors.
+
+If you would prefer the first ``P`` image frame to be ``RGB`` as well, so that
+every ``P`` frame is converted to ``RGB`` or ``RGBA`` mode, there is a setting
+available::
+
+    from PIL import GifImagePlugin
+    GifImagePlugin.PALETTE_TO_RGB = GifImagePlugin.ModeStrategy.ALWAYS
+
+GIF frames do not always contain individual palettes however. If there is only
+a global palette, then all of the colors can fit within ``P`` mode. If you would
+prefer the frames to be kept as ``P`` in that case, there is also a setting
+available::
```

Comment:
> This isn't what DIFFERENT_PALETTE_ONLY does. It still combines the frames. [...]

Oh, then I've misunderstood what `DIFFERENT_PALETTE_ONLY` does. What do you think of changing the part of the docs that threw me?

```suggestion
GIF frames do not always contain individual palettes, however. If there is only
a global palette, then frames may still use values of previous frames, but all
of the colors can fit within ``P`` mode. If you would prefer the frames to be
served as ``P`` in that case, there is also a setting available::
```

Reply:
My hope is that a previous part of the documentation already conveys the idea that new frames are layered on top of old modes.
> ``P`` mode images are changed to ``RGB`` because each frame of a GIF may introduce up to 256 colors. Because ``P`` can only have up to 256 colors, the image is converted to handle all of the colors.

Reply:
Yes, the part that you referenced provides the motivation for why the image is converted to `RGB`. What isn't clear yet (or at least wasn't clear to me when I read it) is that this behavior also applies if we don't convert to `RGB`. Maybe we can rephrase the sentence you referenced to better convey that decoding happens in all cases:

> Transparency allows a GIF frame to use more than 256 colors, 256 from a local palette + other colors from previous frames. To handle this correctly, images will - by default - be promoted to `RGB`. While this recombination (decoding) will always happen, you can control how the result will be returned.

Reply:
I've pushed a commit. What do you think?